spring-core/src/main/java/org/springframework/util/StringUtils.java

```diff
@@ -810,8 +811,9 @@ public static TimeZone parseTimeZoneString(String timeZoneString) {
 		}
 		List<String> result = new ArrayList<>();
 		result.addAll(Arrays.asList(array1));
+		Set<String> stringSet = new HashSet<>(result);
```

Comment:
Sorry, you was right. But HashSet wrapping looks like overengineering (original result.addAll(Arrays.asList(array1)) too :)). Have you tested, how your improvement works on large String arrays? 

---
But you did the same mistake like me) If second array contains duplicate items, you put to  "result" _both_ of duplicates (because your stringSet dont know about new items in result).